src/components/CounterInput/index.d.ts

```diff
@@ -0,0 +1,38 @@
+import {
+    ComponentType,
+    ReactNode,
+    ChangeEvent,
+    MouseEvent,
+    FocusEvent,
+    KeyboardEvent,
+} from 'react';
+import { BaseProps, LabelAlignment } from '../types';
+
+export interface CounterInputProps extends BaseProps {
+    id?: string;
+    name?: string;
+    value?: number;
+    placeholder?: string;
+    tabIndex?: number | string;
+    disabled?: boolean;
+    readOnly?: boolean;
+    required?: boolean;
+    min?: number | string;
+    max?: number | string;
+    isBare?: boolean;
+    error?: ReactNode;
+    step?: number;
+    variant?: 'default' | 'shaded';
+    onChange?: number;
```

Comment:
here I meant that instead of an event we will receive a number, but onChange is function not a number then you need to:
```onChange?: (value: number) => void;```